rudder-core/src/main/scala/com/normation/rudder/services/reports/ExecutionBatch.scala

```diff
@@ -270,10 +272,15 @@ object ExecutionBatch extends Loggable {
    * Takes a string, that should contains a CFEngine var ( $(xxx) or ${xxx} )
    * replace the $(xxx) (or ${xxx}) part by .*
    * and doubles all the \
-   * Returns a string that is suitable for a being used as a regexp
+   * Returns a string that is suitable for a being used as a regexp, with anything not
+   * in ".*" quoted with \Q...\E.
+   * For example, "${foo}(bar)$(baz)foo" => ".*\Q(bar)\E.*\Qfoo\E"
+   *
+   * We also change "\" into "\\" because cfengine escape them that way.
+   *
    */
-  final def replaceCFEngineVars(x : String) : String = {
-    x.replaceAll(replaceCFEngineVars, ".*").replaceAll("""\\""", """\\\\""")
+  final def replaceCFEngineVars(x : String) : Pattern = {
+    Pattern.compile("""\Q"""+ x.replaceAll("""\\""", """\\\\""").replaceAll(replaceCFEngineVars, """\\E.*\\Q""") + """\E""")
```

Comment:
I don't understand the goal of the "\\" => "\\\\" here. Is it because CFEngine change "\" into "\\"? 
If so, why don't we do it *always* (not only when the expression contains a cfengine var), because here, only one of the test on "\\" passe.